rskj-core/src/main/java/co/rsk/util/HexUtils.java

```diff
@@ -223,7 +236,7 @@ public static long jsonHexToLong(String x) {
      * if the parameter has the hex prefix 
      */
     public static boolean hasHexPrefix(final String data) {
-        return data != null && data.startsWith(HEX_PREFIX);
+        return data != null && data.regionMatches(true, 0, HEX_PREFIX, 0, 2);
```

Comment:
why this change is being done? I think the current implementation is doing what it should do.

Reply:
Similar to [this case](https://github.com/rsksmart/rskj/pull/2084#discussion_r1275370029), my aim was to make RSKJ do the same validations as GETH. In this case I saw that GETH is case insensitive when check for the `0x`, this change makes it so RSKJ behaves the same way.